src/test/fixtures/generator.tsx

```diff
@@ -0,0 +1,194 @@
+import { configFileV2, configFileV3 } from "./configuration-file";
+import { ropsten, rinkeby, did } from "./credential";
+import { existsSync, writeFileSync, mkdirSync } from "fs";
+import { join, parse } from "path";
+import { assertConfigFile } from "../../common/config/validate";
+
+const V2_ROPSTEN_DIRECTORY = join(__dirname, "_generated", "v2", "config", "ropsten", "sample-config-ropsten.json");
+const V2_AWS_ROPSTEN_DIRECTORY = join(
+  __dirname,
+  "_generated",
+  "v2",
+  "config",
+  "ropsten",
+  "sample-config-ropsten-aws.json"
+);
+const V2_RINKEBY_DIRECTORY = join(__dirname, "_generated", "v2", "config", "rinkeby", "sample-config-rinkeby.json");
+const V2_DID_DIRECTORY = join(__dirname, "_generated", "v2", "config", "did", "sample-config-did.json");
+
+const V3_ROPSTEN_DIRECTORY = join(__dirname, "_generated", "v3", "config", "ropsten", "sample-config-ropsten.json");
+const V3_RINKEBY_DIRECTORY = join(__dirname, "_generated", "v3", "config", "rinkeby", "sample-config-rinkeby.json");
+const V3_DID_DIRECTORY = join(__dirname, "_generated", "v3", "config", "did", "sample-config-did.json");
+
+function generateConfigFile(configFile: any, version: string) {
+  try {
+    //Ropsten
+    const ropstenConfigFile = {
+      ...configFile,
+      network: ropsten.network,
+      wallet: ropsten.wallet,
+      documentStorage: {
+        ...configFile.documentStorage,
+        url: ropsten.documentStorageUrl,
+      },
+    };
+    if (version === "v2") {
+      const ropstenConfigFileV2 = insertV2Credential(ropstenConfigFile, ropsten);
+      updateConfigFile(ropstenConfigFileV2, V2_ROPSTEN_DIRECTORY);
+    } else {
+      const ropstenConfigFileV3 = insertV3Credential(ropstenConfigFile, ropsten);
+      updateConfigFile(ropstenConfigFileV3, V3_ROPSTEN_DIRECTORY);
+    }
+
+    //AWS Ropsten
+    const ropstenAWSConfigFile = {
+      ...configFile,
+      network: ropsten.network,
+      wallet: ropsten.walletAWS,
+      documentStorage: {
+        ...configFile.documentStorage,
+        url: ropsten.documentStorageUrl,
+      },
+    };
+    if (version === "v2") {
+      updateConfigFile(ropstenAWSConfigFile, V2_AWS_ROPSTEN_DIRECTORY);
+    }
+
+    //Rinkeby
+    const rinkebyConfigFile = {
+      ...configFile,
+      network: rinkeby.network,
+      wallet: rinkeby.wallet,
+      documentStorage: {
+        ...configFile.documentStorage,
+        url: rinkeby.documentStorageUrl,
+      },
+    };
+    if (version === "v2") {
+      const rinkebyConfigFileV2 = insertV2Credential(rinkebyConfigFile, rinkeby);
+      updateConfigFile(rinkebyConfigFileV2, V2_RINKEBY_DIRECTORY);
+    } else {
+      const rinkebyConfigFileV3 = insertV3Credential(rinkebyConfigFile, rinkeby);
+      updateConfigFile(rinkebyConfigFileV3, V3_RINKEBY_DIRECTORY);
+    }
```

Comment:
we can essentially put both v2 and v3 forms in the same sample config yeah? can we put the `name` field in each form as:

TradeTrust Bill of Lading v2
TradeTrust Invoice v2
TradeTrust Chafa Certificate of Origin v2
TradeTrust Bill of Lading v3
TradeTrust Invoice v3
...

Reply:
yup we can, i will do that

Reply:
oh wait, in my current way it can't cause there isn't a way to check which form is v2/ v3. let me think about this further